tsconfig.base.json

```diff
@@ -1,21 +1,54 @@
 {
   "compilerOptions": {
+    "allowJs": false,
     "allowSyntheticDefaultImports": true,
-    "allowUnreachableCode": false,
-    "allowUnusedLabels": false,
+    "composite": true,
     "declaration": true,
     "declarationMap": true,
+    "emitDeclarationOnly": true,
+    "emitDecoratorMetadata": false,
     "esModuleInterop": true,
-    "lib": ["ES2021"],
-    "module": "Node16",
-    "moduleResolution": "Node16",
+    "experimentalDecorators": false,
+    "forceConsistentCasingInFileNames": true,
+    "incremental": true,
+    "isolatedModules": true,
+    "lib": ["es2022"],
+    "module": "NodeNext",
+    "noEmitOnError": true,
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitOverride": true,
     "noImplicitReturns": true,
-    "paths": {},
+    "noUnusedLocals": true,
```

Comment:
can we turn this off?
`no-unused-vars` does this without causing TS errors (so we can remove the `@ts-expect-error`s)
it also means the build won't error out due to a measly unused variable.

Reply:
Cool yeah I just took a relatively strong starting point inspired by other projects, we can change any of it to whatever's best. It isn't something that is generated or controlled by Nx in any way, it's completely up to us